kai/kai_vcr_util.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import os
+from contextlib import contextmanager
+from typing import Generator
+
+import vcr  # type: ignore
+
+from kai.constants import PATH_DATA
+
+KAI_LOG = logging.getLogger("vcr")
+
+
+@contextmanager
+def playback_if_demo_mode(
+    demo_mode: bool, model_id: str, application_name: str, agent: str, filename: str
+) -> Generator[None, None, None]:
+    """
+    A context manager to conditionally use a VCR cassette when demo mode is
+    enabled.
+    """
+
+    record_mode = "once" if demo_mode else "all"
+
+    my_vcr = vcr.VCR(
+        cassette_library_dir=os.path.join(
+            PATH_DATA, "vcr", application_name, agent, model_id
+        ),
+        record_mode=record_mode,
+        match_on=[
+            "uri",
```

Comment:
Are these matchers working for multiple calls for the same file?

Reply:
tbh, I didn't touch the matchers, my assumption was that this was matching based on what we send in http request

Reply:
The matchers were configured for the old use case, if the request body/content changes it will still trigger a match. If you're reading/writing from the same file and only the request body is changing it won't register it as a different request

Reply:
ahh okay, thanks for pointing that out

Reply:
@fabianvf added body to requests for now, seems to be working as expected